Lib/typing.py

```diff
@@ -258,6 +258,16 @@ def _collect_parameters(args):
         # We don't want __parameters__ descriptor of a bare Python class.
         if isinstance(t, type):
             continue
+
+        # `t` might be a tuple, when `ParamSpec` is substituted with
+        # `[T, int]`, or `[int, *Ts]`, etc.
+        if isinstance(t, tuple):
+            for x in t:
+                for collected in _collect_parameters([x]):
+                    if collected not in parameters:
+                        parameters.append(collected)
+            continue
+
         if hasattr(t, '__typing_subst__'):
```

Comment:
I'd find this more readable if it were an `if`/`elif` chain, rather than a series of `continue` statements:

```suggestion
        if isinstance(t, type):
            pass

        # `t` might be a tuple, when `ParamSpec` is substituted with
        # `[T, int]`, or `[int, *Ts]`, etc.
        elif isinstance(t, tuple):
            for x in t:
                for collected in _collect_parameters([x]):
                    if collected not in parameters:
                        parameters.append(collected)

        elif hasattr(t, '__typing_subst__'):
```

Reply:
Agreed, changed.